pkg/csi/cinder/controllerserver.go

```diff
@@ -413,8 +594,18 @@ func (cs *controllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 		return nil, status.Error(codes.InvalidArgument, "Snapshot ID must be provided in DeleteSnapshot request")
 	}
 
+	// If volumeSnapshot object was linked to a cinder backup, delete the backup.
+	back, err := cs.Cloud.GetBackupByID(id)
```

Comment:
Hmm, how do I match for IsNotFound errors? Do i just search for "not found" in `err.Error()`? My environment for testing is temporarily down so if you could help me with this it would be great :pray:

Reply:
Since we can't know if the cinder installation supports backups, I think this is the best method to do it.
Since DeleteBackup() is more error prone than GetBackup().

I think we should first test with GetBackup, and if it does not work, we assume this cinder installation does not support Backups or the backup does not exist. In either case we ignore the error. If it does work but DeleteBackup fails, then we should fail

Reply:
As I explained above, GET call on backups will not tell us anything about `cinder-backup` being present in the deployment, but now I think it does make some sense to probe for backup existence, because on DELETE call we'll crash here: https://opendev.org/openstack/cinder/src/branch/master/cinder/backup/api.py#L120. Thanks, please mark this one as resolved.

Reply:
Oh, interesting, so it's 404 from delete call as well:

```
REQ: curl -g -i --cacert "/home/mdulko/.config/openstack/standalone-ca.crt" -X DELETE https://10.8.1.135:13776/v3/9e1f1eda803b4dd18ec77b3ed4d572a0/backups/eb01d44f-5b1a-4e9e-b23e-7f979bf38dc1 -H "OpenStack-API-Version: volume 3.64" -H "User-Agent: openstacksdk/2.0.0 keystoneauth1/4.5.0 python-requests/2.28.1 CPython/3.10.13" -H "X-Auth-Token: {SHA256}885c7009b71be65f06ddf13031866d95a73b040d50670d7cf7e928ca6eed4914"
https://10.8.1.135:13776 "DELETE /v3/9e1f1eda803b4dd18ec77b3ed4d572a0/backups/eb01d44f-5b1a-4e9e-b23e-7f979bf38dc1 HTTP/1.1" 404 87
RESP: [404] content-length: 87 content-type: application/json date: Thu, 11 Jan 2024 16:42:08 GMT openstack-api-version: volume 3.64 server: Apache vary: OpenStack-API-Version x-compute-request-id: req-f451ec29-8125-4864-9cbb-3476dda8ee3d x-openstack-request-id: req-f451ec29-8125-4864-9cbb-3476dda8ee3d
RESP BODY: {"itemNotFound": {"code": 404, "message": "Service cinder-backup could not be found."}}
DELETE call to block-storage for https://10.8.1.135:13776/v3/9e1f1eda803b4dd18ec77b3ed4d572a0/backups/eb01d44f-5b1a-4e9e-b23e-7f979bf38dc1 used request id req-f451ec29-8125-4864-9cbb-3476dda8ee3d
Failed to delete backup with name or ID 'eb01d44f-5b1a-4e9e-b23e-7f979bf38dc1': ResourceNotFound: 404: Client Error for url: https://10.8.1.135:13776/v3/9e1f1eda803b4dd18ec77b3ed4d572a0/backups/eb01d44f-5b1a-4e9e-b23e-7f979bf38dc1, Service cinder-backup could not be found.
```